extensions/librdkafka/tests/PublishKafkaOnScheduleTests.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file GenerateFlowFile.h
+ * GenerateFlowFile class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#undef NDEBUG
+#include <cassert>
+#include "../../../libminifi/test/integration/IntegrationBase.h"
+#include "core/logging/Logger.h"
+#include "../../../libminifi/test/TestBase.h"
+#include "../PublishKafka.h"
+
+class PublishKafkaOnScheduleTests : public IntegrationBase {
+public:
+    virtual void runAssertions() {
+      std::string logs = LogTestController::getInstance().log_output.str();
+      size_t pos = 0;
+      size_t last_pos = 0;
+      unsigned int occurances = 0;
+      do {
+        pos = logs.find(" value 1 is outside allowed range 1000..1000000000", pos);
```

Comment:
As this logic is not so nice ( I admit to be the guilty one responsible for this :) ), I would prefer not to copy-paste it, but create a function like "countMessagePairsInLog" and make it to be part of test framework. 
If you have a better idea for naming this, I'm happy to see that.

Reply:
Wrapped the logic to a new function